sdk/core/src/policies/transport.rs

```diff
@@ -37,6 +50,184 @@ where
 
         let response = { self.http_client.execute_request2(request) };
 
-        Ok(response.await?.into())
+        Ok(response.await?)
+    }
+
+    #[cfg(feature = "mock_transport_consume")]
+    async fn send(
+        &self,
+        _ctx: &mut PipelineContext<C>,
+        request: &mut Request,
+        next: &[Arc<dyn Policy<C>>],
+    ) -> PolicyResult<Response> {
+        use crate::mock_transport::*;
+        // there must be no more policies
+        assert_eq!(0, next.len());
+
+        // deserialize to file both the request and the response
+        let (read_request, read_response) = {
+            let mut request_path = prepare_and_get_transaction_path();
+            let mut response_path = request_path.clone();
+
+            request_path.push(format!("{}_request.json", get_transaction_num()));
+            response_path.push(format!("{}_response.json", get_transaction_num()));
+
+            let mut request_contents_stream = std::fs::File::open(&request_path)
+                .expect(&format!("Cannot open request file {:?}", request_path));
+
+            let mut request = String::new();
+            request_contents_stream
+                .read_to_string(&mut request)
+                .expect("cannot read request file");
+
+            let mut response_contents_stream = std::fs::File::open(&response_path)
```

Comment:
nit: you can use `std::fs::read_to_string` instead of opening the file and then reading in two steps.